ha/core/controllers/pcs/pcs_controller.py

```diff
@@ -24,4 +32,80 @@ def __init__(self):
+            if fail_count_exists:
+                self.clean_failure_count(node_id)
+            else:
+                resources_healed = True
+                break
+            count += 1
+            time.sleep(10)
+        return resources_healed
+
+    def check_resource_failcount(self, node_id) -> bool:
+        """
+        Resource fail count check
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_FAILCOUNT_STATUS.replace("<node>", node_id),
+                                                   check_error=False)
+        if const.NO_FAILCOUNT in _output:
+            return False
+        else:
+            return True
+
+    def clean_failure_count(self, node_id):
+        """
+        Cleanup resources fail count
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_CLEANUP.replace("<node>", node_id),
+                                                   check_error=False)
+
+    def nodes_status(self, nodeids: list) -> dict:
+        """
+        Get pcs status of nodes.
+        Args:
+            nodeids (list): List of Node IDs from cluster nodes.
+                Default provide list of all node status.
+                if 'local' then provide local node status.
+
+        Returns:
+            ([dict]): Return dictionary. {"node_id1": "status of node_id1",
+                                          "node_id2": "status of node_id2"...}
+        """
+        all_nodes_status = dict()
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_STATUS_NODES, check_error=False)
+        if not isinstance(nodeids, list):
+            raise ClusterManagerError(f"Invalid nodeids type `{type(nodeids)}`, required `list`")
+        for nodeid in nodeids:
+            if nodeid in _output:
+                for status in _output.split("\n"):
+                    nodes = status.split(":")
+                    if len(nodes) > 1 and nodeid.lower() in nodes[1].strip().lower():
+                        if nodes[0].strip().lower() == NODE_STATUSES.STANDBY.value.lower():
+                            all_nodes_status[nodeid] = NODE_STATUSES.STANDBY.value
+                        elif nodes[0].strip().lower() == NODE_STATUSES.STANDBY_WITH_RESOURCES_RUNNING.value.lower():
+                            all_nodes_status[nodeid] = NODE_STATUSES.STANDBY_WITH_RESOURCES_RUNNING.value
+                        elif nodes[0].strip().lower() == NODE_STATUSES.MAINTENANCE.value.lower():
+                            all_nodes_status[nodeid] = NODE_STATUSES.MAINTENANCE.value
+                        elif nodes[0].strip().lower() == NODE_STATUSES.OFFLINE.value.lower():
+                            all_nodes_status[nodeid] = NODE_STATUSES.OFFLINE.value
+                        elif nodes[0].strip().lower() == NODE_STATUSES.ONLINE.value.lower():
+                            all_nodes_status[nodeid] = NODE_STATUSES.ONLINE.value
+                        break
```

Comment:
If the first most node is online in your list, you are updating the dictionary with its status and breaking out of the loop. Right ? Why to break at earliest ?
And if you want this, this can be the first check of your if-else check so that other checks can be skipped.

Reply:
I am breaking the status lines for loop where we are finding the status for the node which is coming from upper for loop.